feature_engine/selection/drop_high_PSI_features.py

```diff
@@ -0,0 +1,460 @@
+from typing import List, Union
+
+import numpy as np
+import pandas as pd
+
+from feature_engine.dataframe_checks import _check_contains_na, _is_dataframe
+from feature_engine.discretisation.equal_frequency import EqualFrequencyDiscretiser
+from feature_engine.discretisation.equal_width import EqualWidthDiscretiser
+from feature_engine.selection.base_selector import BaseSelector
+from feature_engine.variable_manipulation import (
+    _check_input_parameter_variables,
+    _find_or_check_numerical_variables,
+)
+
+Variables = Union[None, int, str, List[Union[str, int]]]
+
+
+class DropHighPSIFeatures(BaseSelector):
+    """
+    DropHighPSIFeatures drops features with a Population Stability Index (PSI) above
+    a given threshold. The PSI of a feature is an indication of the shift in its
+    distribution; a feature with high PSI might therefore be seen as instable.
```

Comment:
I have been thinking about that also as it is not a pure selection such as using correlation. Could we have a module for feature stability?

Reply:
what else would go in a module called feature stability? a module should be something that groups related techniques. A module for 1 technique would be a bit of an overkill.

I wonder if this would fit in a module called "analysis"? l

Reply:
Analysis would definitely be a good option.

Reply:
Yes, I agree, let's keep it in selection, and we just make it very clear in the documentation, what the aim of this selector is, and how it is most commonly used.